src/device.rs

```diff
@@ -535,3 +581,40 @@ pub extern "C" fn wgpu_buffer_get_mapped_range(
 ) -> *mut u8 {
     gfx_select!(buffer_id => GLOBAL.buffer_get_mapped_range(buffer_id, start, size))
 }
+
+/// Fills the given `info` struct with the adapter info.
+///
+/// # Safety
+///
+/// The field `info.name` is expected to point to a pre-allocated memory
+/// location. This function is unsafe as there is no guarantee that the
+/// pointer is valid and big enough to hold the adapter name.
+#[no_mangle]
+pub unsafe extern "C" fn wgpu_adapter_get_info(
+    adapter_id: id::AdapterId,
+    info: &mut CAdapterInfo,
+) -> u8 {
+    let adapter_info = gfx_select!(adapter_id => GLOBAL.adapter_get_info(adapter_id));
+    let adapter_name = match CString::new(adapter_info.name) {
+        Ok(s) => s,
+        Err(_) => return 1,
+    };
+
+    info.device = adapter_info.device;
+    info.vendor = adapter_info.vendor;
+    info.device_type = CDeviceType::from(adapter_info.device_type);
+    info.backend = adapter_info.backend;
+
+    // Copies the string bytes owned into a the pre-allocated memory location
+    // pointed by `info.name`.
+    // NOTE: this is obviousy unsafe and the caller **must** ensure the
+    // memory is allocated.
+    let string_bytes = adapter_name.as_bytes_with_nul();
+    ptr::copy(
+        string_bytes.as_ptr(),
+        info.name as *mut u8,
+        std::cmp::max(info.name_length, string_bytes.len()),
```

Comment:
this should be `min`, but also we need to make sure to add a terminating 0 if we aren't copying the whole `string_bytes`